### PR TITLE
Add new Tests and update Docs

### PR DESCRIPTION
**Pull request by ddspog**
# Proposed Changes

I've added tests for components Auth, Login, PartyRsvp, PartyUninvited, Password and Register, closing issue #6.

I've updated contributing file, with templates and emojis, and removing labels section, closing issue #10. 

I've updated issue and pull request templates, with more information on some sections, closing issue #12.
### Type of changes
- Review Docs (better documentation)
- Testing (automatic tests)
### Features Working

The tests run fast, and all pass. Writing them, allow me to fix errors on some components. 

And also allow me to introduce some code free from testing, making them unreachable during tests.
This was needed because some code only makes logs from console, or makes some states changes. Behavior that wasn't desired for testing, since it overflow the logs.
### Tests

The main tests I made, were to verify initialization of some components variables, and to check call to other methods already tested. They all run and pass from the Mocha Browser tester.
### Docs

I've updated the contributing file, adding some more emojis to put on commit. Added a commit template section. Remove the labels section, and reference the pull request template.

I've also updated the issue templates, guiding to user stories production, and epics goals.

I've also updated the pull request template, removing checkboxes.
### Reviewers:

@ddspog

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/ddspog/mam-seed/pull/13" target="_blank">ReviewNinja

File: imports/ui/components/password/client/password.tests.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            expect(controller.error).to.be.equal('');
+            done();
+        });
+
+        describe('reset()', function() {
+            let validEmail = 'validEmail';
+
+            afterEach(function(done) {
+                if (spies.forgot)
+                    spies.forgot.restore();
```
Comment (ddspog) [inline imports/ui/components/password/client/password.tests.js:55]: Codacy Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684822/issues/source?bid=3340041&fileBranchId=3346427#l55)

File: imports/ui/components/partiesList/client/partiesList.tests.js [hunk 2]
```diff
@@ -5,55 +5,83 @@ import {
...
+import {
+    sinon
+} from 'meteor/practicalmeteor:sinon';
+
 should();
 
 describe('PartiesList', function() {
-    beforeEach(function() {
+    if (!process.env.TESTING)
+        process.env.TESTING = 1;
```
Comment (ddspog) [inline imports/ui/components/partiesList/client/partiesList.tests.js:23]: Codacy Issue found: [Expected '{' and instead saw 'process'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684801/issues/source?bid=3340041&fileBranchId=3346427#l23)

File: imports/ui/components/partiesList/client/partiesList.tests.js [hunk 3]
```diff
@@ -5,55 +5,83 @@ import {
...
+
+    beforeEach(function(done) {
+        if (!stubs.config)
+            stubs.create('config', ConfigGoogleMap, 'apply').returns('');
```
Comment (ddspog) [inline imports/ui/components/partiesList/client/partiesList.tests.js:27]: Codacy Issue found: [Expected '{' and instead saw 'stubs'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684801/issues/source?bid=3340041&fileBranchId=3346427#l27)

File: imports/ui/components/login/client/login.tests.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            done();
+        });
+
+        describe('login()', function() {
+            let validEmail = 'validEmail';
+            let validPassword = 'validPassword';
+
+            afterEach(function(done) {
+                if (spies.login)
+                    spies.login.restore();
```
Comment (ddspog) [inline imports/ui/components/login/client/login.tests.js:54]: Codacy Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684798/issues/source?bid=3340041&fileBranchId=3346427#l54)

File: imports/ui/components/register/client/register.tests.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            done();
+        });
+
+        describe('register()', function() {
+            let validEmail = 'validEmail';
+            let validPassword = 'validPassword';
+
+            afterEach(function(done) {
+                if (spies.register)
+                    spies.register.restore();
```
Comment (ddspog) [inline imports/ui/components/register/client/register.tests.js:60]: Codacy Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684824/issues/source?bid=3340041&fileBranchId=3346427#l60)

File: imports/ui/components/register/client/register.tests.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+import {
+    sinon
+} from 'meteor/practicalmeteor:sinon';
+
+describe('Register', function() {
+    spies.restoreAll();
+
+    if (!process.env.TESTING)
+        process.env.TESTING = 1;
```
Comment (ddspog) [inline imports/ui/components/register/client/register.tests.js:22]: Codacy Issue found: [Expected '{' and instead saw 'process'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684824/issues/source?bid=3340041&fileBranchId=3346427#l22)

File: imports/ui/components/partyAdd/client/partyAdd.tests.js [hunk 7]
```diff
@@ -21,75 +25,114 @@ import {
...
+                    });
+                } else {
+                    controller.submit();
+                    done();
+                }
+            });
+
+            afterEach(function(done) {
+                if (spies.insert)
+                    spies.insert.restore();
```
Comment (ddspog) [inline imports/ui/components/partyAdd/client/partyAdd.tests.js:116]: Codacy Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l116)

File: imports/ui/components/partyAdd/client/partyAdd.tests.js [hunk 8]
```diff
@@ -21,75 +25,114 @@ import {
...
+                if (spies.reset)
+                    spies.reset.restore();
```
Comment (ddspog) [inline imports/ui/components/partyAdd/client/partyAdd.tests.js:118]: Codacy Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l118)

File: imports/ui/components/partyAdd/client/partyAdd.tests.js [hunk 9]
```diff
@@ -21,75 +25,114 @@ import {
 should();
 
 describe('PartyAdd', function() {
+    let user = {
+        username: 'userCreatedName',
+        password: 'userCreatedPassword',
+        create: false,
+        _id: ''
+    }
```
Comment (ddspog) [inline imports/ui/components/partyAdd/client/partyAdd.tests.js:33]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l33)

File: imports/ui/components/partyRsvp/client/partyRsvp.tests.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    sinon
+} from 'meteor/practicalmeteor:sinon';
+
+describe('PartyRsvp', function() {
+    let user = {
+        username: 'userCreatedName',
+        password: 'userCreatedPassword',
+        create: false,
+        _id: ''
+    }
```
Comment (ddspog) [inline imports/ui/components/partyRsvp/client/partyRsvp.tests.js:24]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684816/issues/source?bid=3340041&fileBranchId=3346427#l24)

**Pull request merged**